### PR TITLE
Flow dependencies for groups and flows

### PR DESCRIPTION
**Pull request by ericnewcomer**
* Adds dependencies when using subflows or contact groups in a flow
* Restricts contact group deletion if it is used in a flow add or remove group action
* Restricts flow deletion if it is used in a subflow or a trigger-flow action

screen shot 2017-09-08 at 11 39 22 am

### CODE REVIEW

File: static/less/legacy.less [hunk 1]
```diff
@@ -1969,8 +1969,6 @@ form.smartmin-form ul.errorlist li {
     color: @color-alert-error;
     font-size: 14px;
   }
-
-  .rounded-corners(0);
```
Comment (rowanseymour) [inline static/less/legacy.less]: unrelated?

File: templates/flows/flow_delete.haml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            
+                -plural
+                    There are {{counter}} flows using this flow. It cannot be deleted until it is removed from the
+
+            -for flow in flows
+                <a href="{%url 'flows.flow_editor' flow.uuid%}">{{flow.name}}</a>{{forloop|oxford}}
+            
+            -blocktrans count counter=flows|length
+                flow.
+                -plural
```
Comment (rowanseymour) [inline templates/flows/flow_delete.haml]: indentation doesn't seem quite right - tabs vs spaces ?

File: templates/contacts/contactgroup_delete.haml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    -if flows or triggers
+      -blocktrans with group_name=object.name
+        Sorry, {{group_name}} cannot be deleted quite yet.
+      %p
+
+    -if triggers
+      -blocktrans count counter=triggers|length
+        This group is used by an active trigger. In order to delete it,
+        first remove the
+        -plural
```
Comment (rowanseymour) [inline templates/contacts/contactgroup_delete.haml]: -plural is meant to be at same identation as -blocktrans no? or does it work either way?

File: static/less/legacy.less (hunk 1 above)
Comment (ericnewcomer) [reply static/less/legacy.less]: Ya, kinda, but might as well include it. This gets rid of the little white strip on the left of our dialog headers on Chrome.

File: temba/flows/models.py [hunk 4]
```diff
@@ -2424,6 +2435,41 @@ def get_step_type(dest, rulesets, actionsets):
...
             raise e
 
+    def update_dependencies(self):
+        groups = set()
+        flows = set()
+
+        # find any references in our actions
+        for actionset in self.action_sets.all():
+            for action in actionset.get_actions():
+                if action.TYPE in ('add_group', 'del_group'):
```
Comment (nicpottier) [inline temba/flows/models.py]: maybe use constants here?
